### PR TITLE
🚧 [WIP] Add migration to replace `Link` `external` property

### LINKED ISSUE #8795: [Button and Link quality] [Milestone 1] Create migration to update existing `Link` components to new API
- The migration should include adding underline styles to instances of the `Link` component that are nested in `<p>` tags, `<Text>`, `<Banner>` and footers using `<FooterHelp>`

### PR DESCRIPTION
**Pull request by chazdean**
### WHY are these changes introduced?

Fixes #8795 

### WHAT is this pull request doing?

Creates a migration to update `<Link>` component

- Removes `external` prop and replaces it with `target="_blank"`

### How to 🎩

🖥 [Local development instructions](https://github.com/Shopify/polaris/blob/main/README.md#local-development)
🗒 [General tophatting guidelines](https://github.com/Shopify/polaris/blob/main/documentation/Tophatting.md)
📄 [Changelog guidelines](https://github.com/Shopify/polaris/blob/main/.github/CONTRIBUTING.md#changelog)

<details>
<summary>Copy-paste this code in <code>playground/Playground.tsx</code>:</summary>

```jsx
import React from 'react';
import {Page} from '../src';

export function Playground() {
  return (
    <Page title="Playground">
      {/* Add the code you want to test in here */}
    </Page>
  );
}
```

</details>

### 🎩 checklist

- [ ] Tested on [mobile](https://github.com/Shopify/polaris/blob/main/documentation/Tophatting.md#cross-browser-testing)
- [ ] Tested on [multiple browsers](https://help.shopify.com/en/manual/shopify-admin/supported-browsers)
- [ ] Tested for [accessibility](https://github.com/Shopify/polaris/blob/main/documentation/Accessibility%20testing.md)
- [ ] Updated the component's `README.md` with documentation changes
- [ ] [Tophatted documentation](https://github.com/Shopify/polaris/blob/main/documentation/Tophatting%20documentation.md) changes in the style guide

### CODE REVIEW

**Review by aveline — APPROVED** [review]
L👀ks good Chaz!

File: polaris-migrator/src/migrations/v12-react-replace-link-external-property/tests/v12-react-replace-link-external-property-invalid.input.tsx [hunk 1]
```diff
@@ -0,0 +1,13 @@
+import {Link} from '@shopify/polaris';
+import React from 'react';
+
+export function App() {
+  const condition = false;
+
+  return (
+    <>
+      <Link external={false}>Click Me</Link>
+      <Link external={condition ? 'true' : 'false'}>Click Me</Link>
```
Comment (aaronccasanova) [inline polaris-migrator/src/migrations/v12-react-replace-link-external-property/tests/v12-react-replace-link-external-property-invalid.input.tsx]:
```suggestion
      <Link external={condition ? true : false}>Click Me</Link>
```
Light suggestion

File: polaris-migrator/src/migrations/v12-react-replace-link-external-property/tests/v12-react-replace-link-external-property-invalid.output.tsx [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+
+export function App() {
+  const condition = false;
+
+  return (
+    <>
+      {/* polaris-migrator: Unable to migrate the following expression. Please upgrade manually. */}
+      <Link external={false}>Click Me</Link>
+      {/* polaris-migrator: Unable to migrate the following expression. Please upgrade manually. */}
+      <Link external={condition ? 'true' : 'false'}>Click Me</Link>
```
Comment (aaronccasanova) [inline polaris-migrator/src/migrations/v12-react-replace-link-external-property/tests/v12-react-replace-link-external-property-invalid.output.tsx]:
```suggestion
      <Link external={condition ? true : false}>Click Me</Link>
```

File: polaris-migrator/src/migrations/v12-react-replace-link-external-property/tests/v12-react-replace-link-external-property.input.tsx [hunk 3]
```diff
@@ -0,0 +1,12 @@
+import {Link} from '@shopify/polaris';
+import React from 'react';
+
+export function App() {
+  const condition = false;
+
```
Comment (aaronccasanova) [inline polaris-migrator/src/migrations/v12-react-replace-link-external-property/tests/v12-react-replace-link-external-property.input.tsx]:
```suggestion
```
Doesn't appear to be used in this test

File: polaris-migrator/src/migrations/v12-react-replace-link-external-property/tests/v12-react-replace-link-external-property.output.tsx [hunk 4]
```diff
@@ -0,0 +1,12 @@
+import {Link} from '@shopify/polaris';
+import React from 'react';
+
+export function App() {
+  const condition = false;
+
```
Comment (aaronccasanova) [inline polaris-migrator/src/migrations/v12-react-replace-link-external-property/tests/v12-react-replace-link-external-property.output.tsx]:
```suggestion
```

File: polaris-migrator/src/migrations/v12-react-replace-link-external-property/v12-react-replace-link-external-property.ts [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+  insertJSXComment,
+  removeJSXAttributes,
+} from '../../utilities/jsx';
+import {POLARIS_MIGRATOR_COMMENT} from '../../constants';
+
+export default function v12ReactReplaceLinkComponents(
+  fileInfo: FileInfo,
+  {jscodeshift: j}: API,
+) {
+  const source = j(fileInfo.source);
```
Comment (aaronccasanova) [inline polaris-migrator/src/migrations/v12-react-replace-link-external-property/v12-react-replace-link-external-property.ts]: We'll need to add logic here to resolve internal and external `Link` imports e.g.
```js
// Internal - polaris/polaris-react/src/components/Component.tsx
import {Link} from '../Link'

// External - my-app/src/components/Component.tsx
import {Link} from '@shopify/polaris'
```

Note: We have a utility for that [here](https://github.com/Shopify/polaris/blob/76ed72b41071c55b35e28d2d0859384b3b0da8c6/polaris-migrator/src/migrations/v10-react-replace-text-components/steps/replace-display-text.ts#L52-58)

Comment (aaronccasanova) [reply polaris-migrator/src/migrations/v12-react-replace-link-external-property/v12-react-replace-link-external-property.ts]: We can also exit early if there are [no Polaris `Link` imports in the current file](https://github.com/Shopify/polaris/blob/76ed72b41071c55b35e28d2d0859384b3b0da8c6/polaris-migrator/src/migrations/v10-react-replace-text-components/steps/replace-display-text.ts#L59-L64).

File: polaris-migrator/src/migrations/v12-react-replace-link-external-property/v12-react-replace-link-external-property.ts [hunk 6]
```diff
@@ -0,0 +1,48 @@
...
+
+  const localElementName = 'Link';
```
Comment (aaronccasanova) [inline polaris-migrator/src/migrations/v12-react-replace-link-external-property/v12-react-replace-link-external-property.ts]: We'll need to update this to properly resolve namespaced (local) imports. Here's a [reference pattern](https://github.com/Shopify/polaris/blob/76ed72b41071c55b35e28d2d0859384b3b0da8c6/polaris-migrator/src/migrations/v10-react-replace-text-components/steps/replace-display-text.ts#L66-L68).

**Pull request closed without merging**